radixdlt/src/main/java/com/radixdlt/utils/MemoryLeakDetector.java

```diff
@@ -0,0 +1,87 @@
+package com.radixdlt.utils;
+
+import com.sun.management.GarbageCollectionNotificationInfo;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import javax.management.Notification;
+import javax.management.NotificationEmitter;
+import javax.management.NotificationListener;
+import javax.management.openmbean.CompositeData;
+import java.lang.management.GarbageCollectorMXBean;
+import java.lang.management.ManagementFactory;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class MemoryLeakDetector {
+
+    private static final Logger log = LogManager.getLogger();
+
+    private static final String OldGenGcName = "G1 Old Gen";
+
+    private static final int CONSECUTIVE_INCREASE_ALERT_THRESHOLD = 3;
+
+    private final List<Long> memTrail = new ArrayList<>();
+
+    public void init() {
+        log.info("Starting memory leak detector...");
+        final List<GarbageCollectorMXBean> garbageCollectorMXBeans = ManagementFactory.getGarbageCollectorMXBeans();
+        for (GarbageCollectorMXBean garbageCollectorMXBean : garbageCollectorMXBeans) {
+            listenForGarbageCollectionOn(garbageCollectorMXBean);
+        }
+    }
+
+    private void listenForGarbageCollectionOn(GarbageCollectorMXBean garbageCollectorMXBean) {
+        final NotificationEmitter notificationEmitter = (NotificationEmitter) garbageCollectorMXBean;
+        final GCListener listener = new GCListener();
+        notificationEmitter.addNotificationListener(listener, null, null);
+    }
+
+    private class GCListener implements NotificationListener {
```

Comment:
No real need to wrap this interface in a class.  It could just be a method, and then use a method reference in listenForGarbageCollectionOn(...).

Something like this:
```
    private void listenForGarbageCollectionOn(GarbageCollectorMXBean garbageCollectorMXBean) {
        final NotificationEmitter notificationEmitter = (NotificationEmitter) garbageCollectorMXBean;
        notificationEmitter.addNotificationListener(this::handleNotification, null, null);
    }

    public void handleNotification(Notification notification, Object handback) {
        // ...rest of code here
```